crypto/cmp/cmp_ctx.c

```diff
@@ -0,0 +1,1144 @@
+/*
+ * Copyright 2007-2019 The OpenSSL Project Authors. All Rights Reserved.
```

Comment:
Something doesn't seem right in this preview PR. At least some of the changes also appear in chunk 4, e.g. this file was created by chunk 4, but also seems to appear here.

Reply:
Thanks for notifying us on this issue.
Indeed (the diff of) this PR should only show new source files and additions to existing header files.
It was like this after I updated this PR end of last week, and it looks like some recent attempts to rebase it on the latest changes to chunk 4 (openssl#9107) went wrong.
We'll fix this ASAP.

Reply:
@mattcaswell, finally we've been able to fix the inconsistencies with chunk 4 
such that this PR just shows the new material of chunk 5
(except for a small fix in `cmp_util.c` which I did in both branches but for some weird reason still shows up in the diff here).

So you can resume having a look at this :-)

Reply:
It seems this still has chunk 4 code in it?

Reply:
@mattcaswell, this preview was clean on Saturday, but as soon as any change is pushed to chunk 4, the diff shown here falls apart, so keeping this branch up-to-date is a PITA.
I've just tried to fix this again by rebasing and cherry-picking rescent commits of chunk 4, but in the end the diff showed too much stuff, so I resorted again to building it from scratch.
So currently it's clean again.

Reply:
> @mattcaswell, this preview was clean on Saturday, but as soon as any change is pushed to chunk 4, the diff shown here falls apart, so keeping this branch up-to-date is a PITA.

You probably need to make this PR against a stable branch of chunk 4 - rather than one you are actively working on. That way minor changes to chunk4 should not impact this preview PR.

Reply:
Excellent hint - so I'll freeze the base of this PR (as long as it makes sense).
I hope you can give a first bunch of comments on this preview before your vacation.

Reply:
Will try and take a look tomorrow. Email me if I forget!!

Reply:
Thanks again @mattcaswell for your comments. @Akretsch shall take care of them on his return from vacation.
I've tried doing this branch relative to (some  new version of) the cmpossl_incremental4 branch, but this did not help to avoid stale changes to be shown.

Reply:
I've just reproduced the branch relative to an updated cmpossl_incremental4_base_5 branch;
let's see if now the diff view stays clean.

Reply:
It turns out that also this approach has its issues: 
when there are changes to code of chunk 4 that also affect the code of this preview of chunk 5, cherry-picking them would make the changes to the code of chunk 4 show up in the diff here.
So I'm reverting now to the simpler original approach: making this PR relative to cmpossl_incremental4.